### PR TITLE
Stage imagestreams

### PR DESCRIPTION
**Pull request by alaypatel07**
**Jira**: [MIG-186](https://issues.redhat.com/browse/MIG-186)

This commit modifies the Stage Itinerary to run when either
PV's are found in namespaces or ImageStreams are found in 
namespaces.

It also adds Backup labels to all ImageStream resources to 
backup and restore IS during a staged migration

### CODE REVIEW

File: pkg/controller/migmigration/annotation.go [hunk 1]
```diff
@@ -63,36 +65,38 @@ type ServiceAccounts map[string]map[string]bool
...
 	}
 	// Service accounts used by stage pods.
-	err = t.labelServiceAccounts(client, serviceAccounts)
+	err = t.labelServiceAccounts(sourceClient, serviceAccounts)
 	if err != nil {
 		return liberr.Wrap(err)
 	}
 
+	err = t.labelImageStreams(sourceClient)
+
```
Comment (sseago) [inline pkg/controller/migmigration/annotation.go:99]: We also need to add deleteImageStreamLabels to deleteAnnotations()

**Comment by djwhatle** [comment]
@alaypatel07 please link an associated GH issue or jira

**Comment by alaypatel07** [comment]
@djwhatle I couldn't find a JIRA, will try to coordinate with @jwmatthews to see if there is one or will create one, thanks!

**Comment by sseago** [comment]
@djwhatle @alaypatel07 https://issues.redhat.com/browse/MIG-186 is the relevant jira, although this PR is only item #1 of the 5.

**Comment by djwhatle** [comment]
@alaypatel07 @sseago I created a GH issue to mirror jira and track all of this work, with this PR addressing item 1/5. https://github.com/konveyor/mig-controller/issues/621

**Comment by alaypatel07** [comment]
\test all

**Comment by alaypatel07** [comment]
@sseago @alaypatel07 tested a staged migration, and followed up with a full migration, both tests were green. Final pass at this would be very helpful PTAL.

File: pkg/controller/migmigration/annotation.go (hunk 1 above)
Comment (alaypatel07) [reply pkg/controller/migmigration/annotation.go:99]: added

File: pkg/controller/migmigration/annotation.go [hunk 2]
```diff
@@ -63,36 +65,38 @@ type ServiceAccounts map[string]map[string]bool
...
 		return liberr.Wrap(err)
 	}
 	// Service accounts used by stage pods.
-	err = t.labelServiceAccounts(client, serviceAccounts)
+	err = t.labelServiceAccounts(sourceClient, serviceAccounts)
 	if err != nil {
 		return liberr.Wrap(err)
 	}
 
+	err = t.labelImageStreams(sourceClient)
```
Comment (sseago) [inline pkg/controller/migmigration/annotation.go:98]: Checking the returned `err` is missing here.

File: pkg/controller/migmigration/annotation.go [hunk 3]
```diff
@@ -333,6 +337,38 @@ func (t *Task) labelServiceAccounts(client k8sclient.Client, serviceAccounts Ser
...
+			return err
+		}
+		for _, is := range imageStreamList.Items {
+			if is.Labels == nil {
+				is.Labels = map[string]string{}
+			}
+			is.Labels[IncludedInStageBackupLabel] = t.UID()
+			err = client.Update(context.Background(), &is)
+			if err != nil {
+				// TODO: confirm with the team if we want to collect errors
```
Comment (sseago) [inline pkg/controller/migmigration/annotation.go:357]: I think we'd resolved the collecting errors question for now, so we can probably remove the TODO comment.

File: pkg/controller/migmigration/annotation.go [hunk 4]
```diff
@@ -550,3 +590,31 @@ func (t *Task) deleteServiceAccountLabels(client k8sclient.Client) error {
...
+		err := client.List(context.TODO(), options, &imageStreamList)
+		if err != nil {
+			log.Trace(err)
+			return err
+		}
+		for _, is := range imageStreamList.Items {
+			delete(is.Labels, IncludedInStageBackupLabel)
+			err = client.Update(context.Background(), &is)
+			if err != nil {
+				// TODO: confirm with the team if we want to collect errors
```
Comment (sseago) [inline pkg/controller/migmigration/annotation.go:608]: Remove TODO comment.

File: pkg/controller/migmigration/annotation.go (hunk 2 above)
Comment (alaypatel07) [reply pkg/controller/migmigration/annotation.go]: good catch, resolved

File: pkg/controller/migmigration/annotation.go (hunk 3 above)
Comment (alaypatel07) [reply pkg/controller/migmigration/annotation.go]: added an issue and removed

**Pull request merged**